server/datastore/mysql/hosts.go

```diff
@@ -491,6 +491,12 @@ func (ds *Datastore) applyHostFilters(opt fleet.HostListOptions, sql string, fil
 		params = append(params, opt.SoftwareIDFilter)
 	}
 
+	operatingSystemFilter := "TRUE"
+	if opt.OperatingSystemIDFilter != nil {
+		operatingSystemFilter = "EXISTS (SELECT 1 FROM host_operating_system hos WHERE hos.host_id = h.id AND hos.os_id = ?)"
```

Comment:
I think this subquery will trigger a full table scan on `hosts` because of the functional dependency with the sub-query (`hos.host_id = h.id`) - a better approach here would be do an inner join instead.

Reply:
@juan-fdz-hawa, good eye! The inner join seems much cleaner according to EXPLAIN. It looks like the software filter above may have a similar issue?

Reply:
Yeah